src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultWorkspaceSemanticTokensRefreshPublisher.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class DefaultWorkspaceSemanticTokensRefreshPublisher : WorkspaceSemanticTokensRefreshPublisher
+    {
+        private const string WorkspaceSemanticTokensRefreshKey = "WorkspaceSemanticTokensRefresh";
+        private readonly IClientLanguageServer _languageServer;
+        private BatchingWorkQueue? _workQueue;
```

Comment:
You can mark this as readonly and non-nullable and always create it in the ctor. It wont spin up any asynchronous work unless something has been enqueued onto it. That way you can also remove all of the `?.` calls into it